File: Easy/Problem26.scala
/*
* Runtime: 566ms -> Beats 44.58% of users with Scala
* Memory: 56.14MB -> Beats 69.88% of user with Scala
*/

object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
        if (nums.isEmpty) return 0
  
        var index = 0
        for (i <- 1 until nums.length) {
            if (nums(i) != nums(index)) {
                index += 1
                nums(index) = nums(i)
            }
        }
    
        index + 1
    }
}

File: Easy/Problem26.java
/*
* Runtime: 0ms -> Beats 100% of users with Java
* Memory: 44.68MB -> Beats 63.20% of user with Java
*/

class Solution {
    public int removeDuplicates(int[] nums) {
        int index = 0;
        for(int i=0; i<nums.length; i++){
            if(nums[i]!=nums[index]){
                index++;
                nums[index]= nums[i];
            }
        }
        return index + 1;
    }
}